textbook/chapter_1.qmd

```diff
@@ -0,0 +1,993 @@
+    classDef javascript fill:#FFFFFF,color:#000
+    %% Legend items
+    AT[Alignment Tool]:::alignmentTool
+    HL[High Level]:::highLevel
+    LL[Low Level]:::lowLevel
+    JS[JavaScript]:::javascript
+```
+
+<div class="song-container">
+<div class="song-details">
+<div class="song-icon">🎵</div>
+<div class="song-info">
+<div class="song-title">Data Viz Ecosystem</div>
+<div class="song-artist">Generated with artificial intelligence</div>
+</div>
+</div>
+<div class="audio-player">
+<audio controls>
+<source src="/audio/data-viz-ecosystem.mp3" type="audio/mpeg">
+Your browser does not support the audio element.
+</audio>
+</div>
+</div>
+
+### What is a graphical grammar?
+
+A graphical grammar is a system for creating visualizations by combining basic elements according to specific rules. It parallels the grammars of natural and programming languages, but applies to visual components.
+
+We are familiar with grammatical rules for natural languages,
+where they tell us how to combine words into sentences
+to convey particular meanings.
+
+In programming, different languages have different grammar, for instance if you try to write ```"hello" + "world"``` in Python the grammar of Python defines that the addition of string should perform concatenation so the returned result is `"helloworld"`.
+
+In contrast, the grammar of R says that there is no such thing as addition of strings,
+so it throws an error complaining that non-numeric arguments were passed to the `+` operator.
+
+**Graphical grammar**
+
+Likewise for a visualization grammar,
+different packages make different choices
+for what happens when different visual elements are added together,
+and which those visual elements are.
+
+Key components of a graphical grammar include:
+
+1. **Canvas**: The background or base of the chart.
+2. **Geometric marks**: Visual elements representing data (e.g., bars, dots).
+3. **Visual encoding**: Rules for mapping data to visual properties.
+
+Unlike traditional charting tools that offer pre-made chart types, a grammar-based approach is more flexible. It allows the user to construct custom visualizations from fundamental building blocks.
+
+The following table illustrates the similarities between the grammars of programming, natural language and graphics:
+
+| Component | Natural Language | Programming Language | Graphical Grammar |
+|-----------|-------------------|----------------------|-------------------|
+| Basic Units | Words | Variables, literals | Marks (points, lines, bars) |
+| Syntax Rules | Word order, punctuation | Syntax rules, operators | Rules for combining marks, layers, aesthetics |
+| Semantics | Word meanings, context | Expression meanings | Meanings of visual encodings |
+| Composition | Sentences, paragraphs | Statements, code blocks | Layers, facets |
+| Output | Text | Executable code | Data visualizations |
+| Modifiers | Adjectives, adverbs | Attributes, parameters | Scales, legends, annotations |
+
```

Comment:
Helpful idea with this table. I will need to think a bit more tomorrow on whether I think it is too copmlex, but I like the explicitness

Reply:
I still haven't made up my mind here but left it in for now